src/qibolab/designs/channels.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from qibolab.instruments.abstract import LocalOscillator
+
+
+@dataclass
+class Channel:
+    """Representation of physical wire connection (channel)."""
+
+    name: str
+    """Name of the channel from the lab schematics."""
+
+    qubit: Optional["Qubit"] = field(default=None, repr=False)
+    """Qubit connected to this channel.
+    Used to read the sweetspot and filters for flux channels only. ``None`` for non-flux channels.
+    """
+    ports: List[tuple] = field(default_factory=list)
+    """List of tuples (controller, port) connected to this channel."""
+    local_oscillator: Optional[LocalOscillator] = None
+    """Instrument object controlling the local oscillator connected to this channel.
+    Not applicable for setups that do not use local oscillators because the controller
+    can send sufficiently high frequencies
+    """
+    _offset: Optional[float] = None
+    _filter: Optional[dict] = None
+
+    @property
+    def offset(self):
+        """Bias offset for flux channels."""
+        if self._offset is None:
+            # operate qubits at their sweetspot unless otherwise stated
+            return self.qubit.sweetspot
+        return self._offset
+
+    @offset.setter
+    def offset(self, offset):
+        self._offset = offset
```

Comment:
Shall we place some checks for boundaries or at least types?

Reply:
I added a type check for float. Regarding boundaries, I am not sure what values I should use because I would expect them to be setup specific. For example for QM I've heard from @maxhant that I should avoid setting this outside [-0.2, 0.2] (volts), but for other platforms this should be different.

In fact, depending on the instrument that we are using to apply the DC offset we may have to set current instead of voltage. We have been thinking if we could find some global units that will work for all platforms but in this case translating current to voltage is not trivial at all and I don't think we should go down that road.

Reply:
I understand, we could delegate this assertion to the specific instrument, therefore if the received instruction goes beyond its limits it raises an error.

Reply:
I added a check for the QM offset here https://github.com/qiboteam/qibolab/blob/0f19c4f5812f5b24c394e917a8c16f321193a416/src/qibolab/instruments/qm.py#L232